spyder_kernels/ipdb/tests/test_ipdb_kernel.py

```diff
@@ -0,0 +1,191 @@
+# Based on the tests in the Metakernel package
+# See test_metakernel.py at
+# https://github.com/Calysto/metakernel/metakernel/tests
+#
+# Licensed under the terms of the MIT License
+# (see spyder_kernels/__init__.py for details)
+# -----------------------------------------------------------------------------
+
+"""
+Tests for spyder_kernels.ipdb.kernel.py
+"""
+# Standart library imports
+import os
+
+# Test library imports
+import pytest
+
+# Local imports
+from spyder_kernels.ipdb.kernel import IPdbKernel
+
+# Third party imports
+from metakernel.tests.utils import get_kernel, get_log_text
+
+
+def test_available_magics():
+    """Check the magics available for the kernel."""
+    kernel = get_kernel(kernel_class=IPdbKernel)
+    for magic in ['EOF', 'a', 'activity', 'alias', 'args', 'b', 'break',
+                  'bt', 'c', 'cd', 'cl', 'clear', 'commands', 'connect_info',
+                  'cont', 'continue', 'conversation', 'd', 'disable',
+                  'display', 'dot', 'down', 'download', 'edit', 'enable',
+                  'exit', 'get', 'help', 'html', 'ignore', 'include',
+                  'install', 'install_magic', 'interact', 'j', 'javascript',
+                  'jigsaw', 'jump', 'kernel', 'kx', 'l', 'latex',  'list',
+                  'll', 'load', 'long_list', 'ls', 'lsmagic', 'macro', 'magic',
+                  'matplotlib', 'n', 'next', 'p', 'parallel', 'pdef', 'pdoc',
+                  'pfile', 'pinfo', 'pinfo2', 'plot', 'pmap', 'pp', 'psource',
+                  'px', 'python', 'q', 'quit', 'r', 'reload_magics', 'restart',
+                  'return', 'retval', 'run', 'rv', 's', 'scheme', 'set',
+                  'shell', 'source', 'step', 'tbreak', 'u', 'unalias',
+                  'undisplay', 'unt', 'until', 'up', 'w', 'whatis', 'where']:
+        msg = "magic '%s' is not in line_magics" % magic
+        assert magic in kernel.line_magics, msg
+
+    for magic in ['file', 'html', 'javascript', 'latex', 'shell', 'time']:
+        assert magic in kernel.cell_magics
+
+
+def test_shell_magic():
+    """Test %shell magic."""
+    kernel = get_kernel(kernel_class=IPdbKernel)
+    with open('TEST.txt', 'wb'):
+        pass
+    if os.name != 'nt':
+        kernel.get_magic('%shell ls')
+    else:
+        kernel.get_magic('%shell dir')
+    log_text = get_log_text(kernel)
+    assert 'TEST.txt' in log_text
+    os.remove('TEST.txt')
+
+
+def test_help():
+    """Check availability of help information."""
+    kernel = get_kernel(kernel_class=IPdbKernel)
+    resp = kernel.get_help_on('%shell', 0)
+    assert 'run the line as a shell command' in resp
+
+    resp = kernel.do_execute('%cd?', False)
+    assert 'change current directory of session' in resp[
+        'payload'][0]['data']['text/plain']
+
+    resp = kernel.get_help_on('what', 0)
+    assert resp == ("Sorry, no help is available"
+                    " on 'what'.", ("response was actually %s" % resp))
+
+
+def test_complete():
```

Comment:
This test should also check that trying to complete `imp` should return `import`